projects/components/src/form/number-with-unit-input/number-with-unit-form-input.component.ts

```diff
@@ -185,21 +185,21 @@ export class NumberWithUnitFormInputComponent extends BaseFormControl implements
             });
         } else {
             this.formGroup = this.fb.group({
-                limited: [this.bestValue, validatorUnlimited],
+                limited: [this.bestValue === -1 ? 0 : this.bestValue, validatorUnlimited],
                 comboUnitOptions: this.bestUnit.getMultiplier(),
-                unlimited: false,
+                unlimited: this.bestValue === -1 ? true : false,
             });
             this.tracker.subscribe(this.formGroup.get('unlimited').valueChanges, value => {
                 const input = this.formGroup.get('limited');
                 const comboUnitOptions = this.formGroup.get('comboUnitOptions');
-                if (value) {
-                    input.disable();
-                    comboUnitOptions.disable();
-                } else {
-                    input.enable();
-                    comboUnitOptions.enable();
+                // If checkbox is limited and value is -1, or if checkbox is unlimited and input value is empty,
+                // set input value to 0.
+                if ((value === false && input.value === -1) || (value === true && input.value)) {
+                    input.setValue(0);
                 }
-                this.onChange(this.value());
+                const inputValue = this.value();
+                this.formControl.setValue(inputValue ? inputValue : 0, { emitModelToViewChange: true });
+                // this.onChange(this.value());
```

Comment:
Don't leave uncommented code in. Please remove it.

Right now, it's a bit unclear as to why we're doing something different here. We could add comments to explain but we could also rework this.

Bring back `this.onChange(this.value)`

After calling `this.onChange`, we call a new private method that enabled/disables fields based on whether it's unlimited. That new function will be  called from `writeValue()` and from here.

Example:

```ts

// Makes sure that the input fields are disabled when this widget is set to UNLIMITED
private updateUiDisabledState() {
        if (value === UNLIMITED) {
            input.disable();
            comboUnitOptions.disable();
        } else {
            input.enable();
            comboUnitOptions.enable();
            input.setValue(this.bestValue);
            comboUnitOptions.setValue(this.bestUnit.getMultiplier());
        }
}
```


```